src/NuGet.Core/NuGet.Packaging/LocalPackageArchiveDownloader.cs

```diff
@@ -144,23 +144,20 @@ public void Dispose()
 
             cancellationToken.ThrowIfCancellationRequested();
 
+            // This value is copied from System.IO.Stream.
+            // It is the largest multiple of 4096 that is smaller than the large object heap (LOH)
+            // threshold of 85K and offers better copy performance than a 4K buffer.
+            const int bufferSize = 81920;
+
             using (var destination = new FileStream(
                 destinationFilePath,
                 FileMode.Create,
-                FileAccess.ReadWrite,
-                FileShare.ReadWrite | FileShare.Delete,
-                bufferSize: 4096,
+                FileAccess.Write,
+                FileShare.Write,
+                bufferSize,
                 useAsync: true))
```

Comment:
do you see better perf when setting useAsync to false? I recall a perf issue in the past where using async caused additional security checks each time a different thread resumed the operation, and the recommendation was to just not use async file streams. @alpaix would know better.